### PR TITLE
Implement

### PR DESCRIPTION
**Pull request by tomrf1**
Lambda to be triggered by s3 object creation events.
If the object is below the specified max size, copy it to the specified destination bucket.
If the object is above max size, split it into files of ~max size in the destination bucket.

### CODE REVIEW

File: src/lambda.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        reader.on('line', (line: string) => {
+            outputStream.write(line + '\n');
+
+            if (outputStream.bytesWritten > MaxKb * 1000) {
+                //Begin writing to a new file stream before closing the old one
+                let oldOutputStream = outputStream;
+                outputStream = fs.createWriteStream(`/tmp/${buildFileName(fileNumber + 1)}`);
+
+                oldOutputStream.end();
+                uploadToS3(fileNumber++);
```
Comment (regiskuckaertz) [inline src/lambda.ts:78]: this is equivalent to 
```js
uploadToS3(fileNumber)
filterNumber += 1
```
is this what you want?

Comment (tomrf1) [reply src/lambda.ts:78]: yes, it uploads the old (completed) file to s3, then increments the counter

**Comment by regiskuckaertz** [comment]
It looks good 👍  but I think you can do away with the `Readline` and use the `data` event instead, as it will have a negative impact on performance.

**Comment by tomrf1** [comment]
Readline is useful because the new files must end with a full line, whereas I believe if I just used the `data` event I'd essentially have to re-implement Readline

**Comment by regiskuckaertz** [comment]
Yes indeed. It's a shame they do not provide a transformer for that by default, because then your process becomes
```js
readFromS3(record).pipe(chunk(maxBytes)).pipe(writeToS3(record))
```
and all the complexity goes away

**Comment by tomrf1** [comment]
that would be nice, though we'd still need to create a new stream for each new s3 file

**Comment by regiskuckaertz** [comment]
yes, `chunk: int => Transform` and `writeToS3: Record => Writeable`, and each provides a nice decoupling.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,4 @@
+dist
+target
+node_modules
+.idea/
```
Comment (regiskuckaertz) [inline .gitignore:4]: tip: you can place that line in a global `.gitignore` ([see](http://egorsmirnov.me/2015/05/04/global-gitignore-file.html)) and never have to put it into a project again ✌️

File: package.json [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "s3-chunking-lambda",
+  "isAwsLambda": true,
+  "cloudformation": false,
+  "projectName": "Off Platform::s3-chunking-lambda",
+  "buildDir": "./target",
+  "riffraffFile": "./riff-raff.yaml",
+  "dependencies": {
+    "aws-sdk": "^2.264.1",
+    "typescript": "2.8.3"
```
Comment (regiskuckaertz) [inline package.json:10]: Great choice 😎

**Pull request merged**